pkg/subctl/cmd/validate_cni.go

```diff
@@ -73,5 +86,77 @@ func validateCniConfig(cmd *cobra.Command, args []string) {
 			submariner.Status.NetworkPlugin)
 		status.QueueSuccessMessage(message)
 		status.End(cli.Success)
+
+		if submariner.Status.NetworkPlugin == "canal-flannel" {
+			validateIPPool(item.config, submariner)
+		}
+	}
+}
+
+func validateIPPool(config *rest.Config, submariner *v1alpha1.Submariner) {
+	dynClient, err := dynamic.NewForConfig(config)
+	exitOnError("Error creating clients for cluster", err)
+
+	client := dynClient.Resource(calicoGVR)
+
+	if err != nil {
+		exitOnError("Error creating Kubernetes client", err)
+	}
+
+	message := "Validating that Calico IPPools configuration"
+	status.Start(message)
+
+	findClusterCIDR := false
+	findServiceCIDR := false
+	ippools, err := client.List(metav1.ListOptions{})
+	if err != nil {
+		message := fmt.Sprintf("Error obtaining IPPools: %v", err)
+		status.QueueFailureMessage(message)
+		status.End(cli.Failure)
+	}
+
+	if len(ippools.Items) < 1 {
+		message := "IPPool is missing"
+		status.QueueFailureMessage(message)
+		status.End(cli.Failure)
+	}
+
+	for _, pool := range ippools.Items {
+		cidr, found, err := unstructured.NestedString(pool.Object, "spec", "cidr")
+
+		if err != nil {
+			message := fmt.Sprintf("Error extracting field cidr from IPPool: %v", pool.GetName())
+			status.QueueFailureMessage(message)
+			status.End(cli.Failure)
+		}
+
+		if !found {
+			message := fmt.Sprintf("No cidr found for IPPool: %v", pool.GetName())
+			status.QueueFailureMessage(message)
+			continue
+		}
+
+		if cidr == submariner.Status.ClusterCIDR {
+			findClusterCIDR = true
+		} else if cidr == submariner.Status.ServiceCIDR {
+			findServiceCIDR = true
+		}
+	}
+
+	if !findClusterCIDR {
+		message := fmt.Sprintf("IPPool with cluster CIDR is not set. (%v)", submariner.Status.ClusterCIDR)
+		status.QueueFailureMessage(message)
 	}
+	if !findServiceCIDR {
+		message := fmt.Sprintf("IPPool with service CIDR is not set. (%v)", submariner.Status.ServiceCIDR)
+		status.QueueFailureMessage(message)
+	}
+
+	if !findServiceCIDR || !findClusterCIDR {
+		status.End(cli.Failure)
+	}
+
+	message = "All Calico IPPools are configured according to cluster CIDR and service CIDR"
+	status.QueueSuccessMessage(message)
+	status.End(cli.Success)
```

Comment:
I don't think we need this final success message - the green check on the "Validating ..." message should suffice.

```suggestion
	status.End(status.ResultFromMessages())
```